subsys/bluetooth/host/hci_core.c

```diff
@@ -4527,3 +4459,41 @@ int bt_configure_data_path(uint8_t dir, uint8_t id, uint8_t vs_config_len,
 
 	return err;
 }
+
+/* Return `true` if a command was processed/sent */
+static bool process_pending_cmd(k_timeout_t timeout)
+{
+	if (!k_fifo_is_empty(&bt_dev.cmd_tx_queue)) {
+		if (k_sem_take(&bt_dev.ncmd_sem, K_NO_WAIT) == 0) {
```

Comment:
Since we are doing `K_NO_WAIT` here, why do we need the `if (!k_fifo_is_empty(&bt_dev.cmd_tx_queue))`?

Reply:
not the same resource. We check if we have pending commands and only then do we take the command lock/resource with `K_NO_WAIT`